### PR TITLE
UIIN-2193 Define new route for Inventory "Browse" page

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/UIIN-2193
Add a new route to move the "Browse" logic to a separate page.
## Approach

- Define a new route `inventory/browse`
- Add `stripes-acq-components` to deps (To reuse existing hooks, components and utils)
- Create base components for "Browse" functionality (For now it's just a blank page with two panes)
- Create toggle component (`<SearchModeNavigation>`) to switch between "Search" and "Browse" modes
- Add `<SearchModeNavigation>` to "Search" and "Browse" pages
- Add unit test to new components

<details>
  <summary>Preview</summary>
  
newBrowsePAge

  
</details>

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by usavkov-epam** [comment]
@folio-org/fe-tl-reviewers review please, thx

**Review by mkuklis — APPROVED** [review]
This looks excellent @usavkov-epam. Nice work!

The only question I have is will the previous setup be cleaned up eventually?

**Comment by usavkov-epam** [comment]
> This looks excellent @usavkov-epam. Nice work!
> 
> The only question I have is will the previous setup be cleaned up eventually?

Yes, after moving "Browse" logic, the old code will be cleaned up

File: package.json [hunk 1]
```diff
@@ -881,6 +881,7 @@
   },
   "dependencies": {
     "@folio/quick-marc": "^5.0.0",
+    "@folio/stripes-acq-components": "~3.2.0",
```
Comment (NikitaSedyx) [inline package.json:884]: don't forget to add to `stripesDeps`

File: src/components/InstancesList/InstancesList.js [hunk 2]
```diff
@@ -298,7 +299,10 @@ class InstancesList extends React.Component {
   }
 
   renderNavigation = () => (
-    <FilterNavigation segment={this.props.segment} onChange={this.refocusOnInputSearch} />
+    <>
+      <SearchModeNavigation />
```
Comment (NikitaSedyx) [inline src/components/InstancesList/InstancesList.js:303]: I suggest to not show this navigation on instances list for now as there is a chance we won't complete whole feature in Nolana

Comment (usavkov-epam) [reply src/components/InstancesList/InstancesList.js]: Yes, it's reasonable. I commented out this line until Browse is refactored

File: package.json (hunk 1 above)
Comment (usavkov-epam) [reply package.json]: Done

File: src/components/BrowseResultsPane/BrowseResultsPane.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  const paneSub = <FormattedMessage id="ui-inventory.title.subTitle.browseCall" />;
+
+  const firstMenu = useMemo(() => (
+    isFiltersOpened
+      ? null
+      : (
+        <PaneMenu>
+          <ExpandFilterPaneButton
+            filterCount={getFiltersCount(filters)}
```
Comment (vashjs) [inline src/components/BrowseResultsPane/BrowseResultsPane.js:35]: please chech if `filters` var should be included in dependency array

**Review by vashjs — APPROVED** [review]

File: src/components/BrowseResultsPane/BrowseResultsPane.js (hunk 3 above)
Comment (usavkov-epam) [reply src/components/BrowseResultsPane/BrowseResultsPane.js:35]: Good catch, thanks!

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**